src/app/main/main.controller.js

```diff
@@ -201,6 +201,61 @@
             }
             return true;
         };
+      
+      // Custom Widget Code Starts here
+        
+        vm.openCustomWidgetModal = function(){
+          
+            var customWidgetModal = {
+                backdrop: true,
+                keyboard: true,
+                modalFade: true,
+                scope: $rootScope,
+                templateUrl: 'app/components/modal/customWidgetModal.html',
+                controller: 'MainCtrl',
+                resolve: {
+                  metricsMetadata: function() {
+                    return $rootScope.metricsMetadata
+                  }
+                }
+            };
+
+            var customWidgetModalOptions = {
+                closeButtonText: 'Cancel',
+                actionButtonText: 'Ok',
+                headerText: 'Reset Dashboard',
+                bodyText: 'Are you sure you want to reset the dashboard?'
+            };
+
+            ModalService.showModal(customWidgetModal, {}).then(function() {
+                
+            });
+        };
+      
+        vm.customWidget = {};
+      
+        vm.customWidget.customWidgetOptions = {
```

Comment:
Assuming this is static now because it's not pulling from the input fields yet, correct?